### PR TITLE
fixed employee expense result dropdown

### PR DESCRIPTION
**Pull request by merlouduron**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [x] Fix dynamic dropdown on Employee Expense Results Registrations page
    - [x] Removed logs
    - [x] Fix Project Sales Registration search function bug
    - [x] Fixed both normal employee expense and employee expense results registration

## Link to Backlog Ticket
[https://dattjapan.backlog.com/view/BOP_MANAGEMENT_SYSTEM-163](https://dattjapan.backlog.com/view/BOP_MANAGEMENT_SYSTEM-163)

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 1]
```diff
@@ -461,9 +476,9 @@ const EmployeeExpensesRegistration = () => {
                                   onChange={(e) => handleInputChange(containerIndex, projectIndex, e)}
                                 >
                                   <option value=''></option>{' '}
-                                  {months.map((month, idx) => (
-                                    <option key={idx} value={month}>
-                                      {language === 'en' ? monthNames[month].en : monthNames[month].jp}{' '}
+                                  {projects.map((month, idx) => (
```
Comment (edward-stalley-work) [inline client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:479]: As discussed in the meeting.
We need to make sure the month and year display only for dates that exist for that project.

**Review by edward-stalley-work — COMMENTED** [review]
Same content as the meeting.
Just leave a done comment when finished!

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 2]
```diff
@@ -19,11 +19,19 @@ import {
   getFieldChecks,
   checkForDuplicates,
 } from '../../utils/validationUtil'
-
-
-const months = [
-  '4', '5', '6', '7', '8', '9', '10', '11', '12', '1', '2', '3'
-];
+import { getFilteredEmployeeExpense } from '../../api/EmployeeExpenseEndpoint/FilterGetEmployeeExpense';
```
Comment (edward-stalley-work) [inline client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:22]: on HandleSubmit I am getting an error and cannot successfully create any records.
I tried with multiple projects and dates and it always returned an error.

Do I need to delete migrations etc?

Screenshot 2024-12-17 at 17 45 41

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please See comments

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 1 above)
Comment (merlouduron) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx]: Done.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 2 above)
Comment (merlouduron) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:22]: Apologies for this one. The payload is different so I made some adjustments on `views.py`. It should be working now. 
Fixed.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 3]
```diff
@@ -405,7 +405,7 @@ const EmployeeExpensesRegistration = () => {
 
   const handleSubmit = async (e) => {
     e.preventDefault()
-
+
```
Comment (edward-stalley-work) [inline client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:408]: can delete this added space

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx [hunk 4]
```diff
@@ -456,7 +456,8 @@ const EmployeeExpensesRegistration = () => {
       setCrudValidationErrors([])
     }
     // Continue with submission if no errors
-
+    console.log('employeeContainers', employeeContainers)
+
```
Comment (edward-stalley-work) [inline client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:460]: can delete this console log

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 2 above)
Comment (edward-stalley-work) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:22]: Thank you - it works now.
However I get same error on results screen.
The same changes may need to be made there as well.

Screenshot 2024-12-18 at 09 22 59

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Please see new comments.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 3 above)
Comment (merlouduron) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:408]: Fixed.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 4 above)
Comment (merlouduron) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:460]: Fixed.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 2 above)
Comment (merlouduron) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx:22]: Fixed.

File: client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx (hunk 3 above)
Comment (edward-stalley-work) [reply client/src/pages/EmployeeExpenses/EmployeeExpensesRegistration.tsx]: the whitspace still exists (green area)
Need to use backspace until cursor is at left side!

**Review by edward-stalley-work — APPROVED** [review]

**Pull request merged**